### PR TITLE
docs(stack): include more documentation about end-to-end tests

### PR DESCRIPTION
**Pull request by edgarulg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
 
 ## End-to-end tests
 
-When Astrolabe opens a PR against this repo, we [deploy the updated service to our staging environment](https://spinnaker-staging.cloud.armory.io/#/applications/stagingcd/executions?pipeline=Deploy%20Spinnaker%20into%20SaaS%20Staging). We then [run our pipeline-based end-to-end tests](https://spinnaker-staging.cloud.armory.io/#/applications/armoryhellodeploy/executions?pipeline=Integration-Test-Runner) against the updated Spinnaker.
+When Astrolabe opens a PR against this repo we treat it as a Release Candidate (RC), so we run an End to End test suit to validate the new service version against the current stack.
+
+### Armory End-to-end Pipelines
+
+The `Deploy Test Staging` job is responsible to deploy the new service version into SaaS staging, the SaaS staging environment has a Spinnaker instance with the service versions described in the [stack.yml](https://github.com/armory-io/armory-stack/blob/master/stack.yml).
```
[the remark this answers is not in the document]
Comment (danielpeach) [reply README.md:17]: Would put a period after "the new service version into Saas staging."

File: README.md [hunk 2]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+The configuration for the SaaS staging env is located in the [spinnaker-saas](https://github.com/armory-io/spinnaker-saas/tree/master/hosted-core-infrastructure/spinnaker-staging) repo.
+
+We use a Spinnaker [pipeline](https://spinnaker-staging.cloud.armory.io/#/applications/stagingcd/executions?pipeline=Deploy%20Spinnaker%20into%20SaaS%20Staging) in staging to deploy the new service version.
+If the new service version is deployed successfully, we then [run our pipeline-based end-to-end tests](https://spinnaker-staging.cloud.armory.io/#/applications/armoryhellodeploy/executions?pipeline=Integration-Test-Runner) against the updated Spinnaker.
+
+#### Images & Commits
+
+In the LTS release, every commit in the Armory extension projects generates a new Docker image, Astrolabe opens a PR with the new Docker image & the commit that the image belongs to.
```
Comment (danielpeach) [inline README.md:25]: Would add a period after "generates a new Docker image".

File: README.md [hunk 3]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+Also, we can take a look to the pods in the cluster to check status & logs by the service changed. You can connect to the SaaS staging using this [kubeconfig](https://github.com/armory-io/spinnaker-saas/blob/master/hosted-core-infrastructure/spinnaker-staging/kubeconfig). Under `spinnaker` namespace.
+
+```shell script
+kubectl --kubeconfig=cloud-staging-kube -n spinnaker get pods
+```
+
+If the end-to-end tests failed, we only can check which test failed searching in the end-to-end pipeline. To identify executions in Spinnaker we send as payload the serviceName, image & tag.
+You can take the logs for the service using `kubectl`
+
+To retry the check, we need to re-run the [Githbub Action](https://github.com/armory-io/armory-stack/pull/333/checks?check_run_id=3003102552).
```
Comment (danielpeach) [inline README.md:60]: s/Githbub/Github

File: README.md [hunk 4]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+
+The Armory extension projects uses an OSS Service as base, currently the extension projects uses a LTS version that represents a commit in the OSS. A base artifact can be found [here](https://github.com/armory-io/orca-armory/blob/a08ea272f752ebcce01e59e790abbe0381b10d00/gradle.properties#L2) in the extension project. The base service version is maintained by Astrolabe.
+
+The base artifacts points to a commit in the OSS projects but those artifact are [jars](https://armory.jfrog.io/ui/repos/tree/General/gradle-dev-local%2Fio%2Fspinnaker) or npm packages in Artifactory. The [build ID](https://armory.jfrog.io/ui/repos/tree/Builds/gradle-dev-local%2Fio%2Fspinnaker%2Forca%2Forca-core%2F2021.07.07.21.31.05.master%2Forca-core-2021.07.07.21.31.05.master.jar) of the artifact is the commit sha from the OSS project.
+
+#### Deploy Pipeline
+
+We use Spinnaker to upgrade itself, the `Deploy RC to Staging` workflow enqueue PR checks, so we deploy & test a new service version at a time
```
Comment (danielpeach) [inline README.md:34]: "We use Spinnaker to upgrade itself using the `Deploy RC to Staging` workflow enqueue PR checks so we deploy & test one new service version at a time."

File: README.md [hunk 5]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+
```
Comment (danielpeach) [inline README.md:35]: "it patch" -> "it patches"

File: README.md [hunk 6]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+The [pipeline](https://spinnaker-staging.cloud.armory.io/#/applications/stagingcd/executions?pipeline=Deploy%20Spinnaker%20into%20SaaS%20Staging) uses a [CLI](https://github.com/armory-io/saas-tools) to extract the k8s manifests for the service changed. Then, it patch the new image tag to use and apply the manifests in the staging env. it takes the [configurations](https://github.com/armory-io/spinnaker-saas/blob/master/hosted-core-infrastructure/spinnaker-staging/kustomization.yml) from SaaS staging env.
+
+#### End-to-end pipeline tests
+
+The [pipelines](https://spinnaker-staging.cloud.armory.io/#/applications/armoryhellodeploy/executions?pipeline=Integration-Test-Runner) are the same end-to-end test that we use in preprod but we stil are migrating them. A single pipeline is responsible of trigger all the test.
```
Comment (danielpeach) [inline README.md:40]: "same end-to-end tests"

Comment (danielpeach) [reply README.md:40]: "stil" -> "still"

Comment (danielpeach) [reply README.md:40]: "is responsible for triggering all of the tests"

File: README.md [hunk 7]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+#### Deploy Pipeline
+
+We use Spinnaker to upgrade itself, the `Deploy RC to Staging` workflow enqueue PR checks, so we deploy & test a new service version at a time
+
+The [pipeline](https://spinnaker-staging.cloud.armory.io/#/applications/stagingcd/executions?pipeline=Deploy%20Spinnaker%20into%20SaaS%20Staging) uses a [CLI](https://github.com/armory-io/saas-tools) to extract the k8s manifests for the service changed. Then, it patch the new image tag to use and apply the manifests in the staging env. it takes the [configurations](https://github.com/armory-io/spinnaker-saas/blob/master/hosted-core-infrastructure/spinnaker-staging/kustomization.yml) from SaaS staging env.
+
+#### End-to-end pipeline tests
+
+The [pipelines](https://spinnaker-staging.cloud.armory.io/#/applications/armoryhellodeploy/executions?pipeline=Integration-Test-Runner) are the same end-to-end test that we use in preprod but we stil are migrating them. A single pipeline is responsible of trigger all the test. 
+if one of them ends with TERMINAL status, all the other pipelines stops.
```
Comment (danielpeach) [inline README.md:41]: "If one of them ends with TERMINAL status, all the other pipelines stop."

File: README.md [hunk 8]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+
+#### Troubleshooting
+
+When the PR check failed there could be three reasons why it's failing:
```
Comment (danielpeach) [inline README.md:45]: When a PR check falls

File: README.md [hunk 9]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+- it fails when deploying.
+- An end-to-end pipeline failed.
+- A false alarm.
+
+The deploy & test happened in different steps in the action, if the deploy failed we can get more information in the [deploy pipeline](https://spinnaker-staging.cloud.armory.io/#/applications/stagingcd/executions?pipeline=Deploy%20Spinnaker%20into%20SaaS%20Staging).
```
Comment (danielpeach) [inline README.md:50]: Deploying and testing happen in different action steps.

File: README.md [hunk 10]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+Also, we can take a look to the pods in the cluster to check status & logs by the service changed. You can connect to the SaaS staging using this [kubeconfig](https://github.com/armory-io/spinnaker-saas/blob/master/hosted-core-infrastructure/spinnaker-staging/kubeconfig). Under `spinnaker` namespace.
```
Comment (danielpeach) [inline README.md:51]: we can take a look at the pods

Comment (danielpeach) [reply README.md:51]: "Use the `spinnaker` namespace"

File: README.md [hunk 11]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+
+```shell script
+kubectl --kubeconfig=cloud-staging-kube -n spinnaker get pods
+```
+
+If the end-to-end tests failed, we only can check which test failed searching in the end-to-end pipeline. To identify executions in Spinnaker we send as payload the serviceName, image & tag.
```
Comment (danielpeach) [inline README.md:57]: "we can only check which test failed by searching in the end-to-end pipeline"

File: README.md [hunk 12]
```diff
@@ -10,7 +10,56 @@ See the [release FAQ](https://go.armory.io/release-faq).
...
+You can take the logs for the service using `kubectl`
```
Comment (danielpeach) [inline README.md:58]: Do we have logs for the staging environment in New Relic? It would be easier to point people to these instead of using kubectl.

**Review by danielpeach — APPROVED** [review]

File: README.md (hunk 12 above)
Comment (edgarulg) [reply README.md:58]: I agree, but I didn't see a config for that. I would like to send those logs to New Relic give a better experience to the Engineers.

**Pull request merged**